plugins/modules/zabbix_user_role.py

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2022, mrvanes
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+module: zabbix_user_role
+short_description: Adds or removes zabbix roles
+author:
+    - Martin van Es
+description:
+    - This module adds or removes zabbix roles
+requirements:
+    - "python >= 2.6"
+    - "zabbix-api >= 0.5.4"
+options:
+    state:
+    name:
+        description: Name of the role to be processed
+    type:
+        description:
+            - User (default)
+            - Admin
+            - Super admin
+    rules:
+        - description: Rules set as defined in https://www.zabbix.com/documentation/current/en/manual/api/reference/role/object#role-rules
```

Comment:
I would suggest this to be covered fully in the module as seen in the [zabbix_user](https://github.com/ansible-collections/community.zabbix/blob/eecef88328d7b80eae1973a2dfcf232624506bd9/plugins/modules/zabbix_user.py#L111-L174) module for the example.

At minimum, it needs:
```
type: dict
required: true/false
default: {}
```

I am not sure however if this will satisfy sanity tests (enforced by the upstream - Ansible) and `suboptions` block may be required.